client/sftp_client.py

```diff
@@ -196,202 +614,749 @@ def load_dirs(self, path, parent=None):
+				owner = True
 		perm = '   '
-		_file = self.ftp.file(fullname)
-		perm += 'r' if _file.readable() else '-'
-		perm += 'w' if _file.writable() else '-'
-		_file.close()
-		perm += 'x' if bool(int(oct(mode)[2:]) & stat.S_IXOTH) else '-'
+		if other:
+			perm += 'r' if bool(mode & stat.S_IROTH) else '-'
+			perm += 'w' if bool(mode & stat.S_IWOTH) else '-'
+			perm += 'x' if bool(mode & stat.S_IXOTH) else '-'
+		elif group:
+			perm += 'r' if bool(mode & stat.S_IRGRP) else '-'
+			perm += 'w' if bool(mode & stat.S_IWGRP) else '-'
+			perm += 'x' if bool(mode & stat.S_IXGRP) else '-'
+		elif owner:
+			perm += 'r' if bool(mode & stat.S_IRUSR) else '-'
+			perm += 'w' if bool(mode & stat.S_IWUSR) else '-'
+			perm += 'x' if bool(mode & stat.S_IXUSR) else '-'
+		else:
+			raise ValueError('Permission not parsed correctly')
 		return perm
+	###FIX ME
 
 	def _yield_dir_list(self, path):
 		for name in self.ftp.listdir(path):
 			yield name
 
+	def _already_exists(self, path):
+		try:
+			self.ftp.stat(path)
+		except IOError as error:
+			if error[0] == 2:
+				return False
+			else:
+				 raise error
+		else:
+			return True
+
+	def _rename_file(self, _iter, path):
+		self.ftp.rename(self._tv_model[_iter][2], path)  #pylint: disable=unsubscriptable-object
+
+	@handle_permission_denied
+	def _make_file(self, path, ftp=None):
+		# If method called when self.ftp is busy, alternate ftp is specified, prevents Garbage Packet Error
+		if ftp is None:
+			ftp = self.ftp
+		ftp.mkdir(path)
+
 	def _get_raw_time(self, fullname):
-		lstatout = self.ftp.lstat(fullname)
+		lstatout = self.ftp.stat(fullname)
 		return lstatout.st_mtime
 
-	def _get_is_folder(self, fullname):
-		lstatout = self.ftp.lstat(fullname)
-		mode = lstatout.st_mode
-		return stat.S_ISDIR(mode)
+	@handle_permission_denied
+	def delete(self, model, treeiter):
+		name = self._tv_model[treeiter][2]
+		if self.get_is_folder(name):  #pylint: disable=unsubscriptable-object
+			if not self.remove_by_folder_name(name):
+				return
+		else:
+			if not self.remove_by_file_name(name):
+				return
+		gui_utilities.show_dialog_warning('Successfully deleted ' + name, self.application.get_active_window())
+		self.refresh()
+		if self.parentless:
+			self._tv_model.clear()
+			self.load_dirs(self.default_directory)
+
+	###FIX ME
+	@handle_permission_denied
+	def remove_by_folder_name(self, name):
+		# with paramiko, you cannot remove populated dir, so recursive method utilized
+		for path in self._yield_dir_list(name):
+			new_path = os.path.join(name, path)
+			if self.get_is_folder(new_path):
+				self.remove_by_folder_name(new_path)
+			else:
+				self.remove_by_file_name(new_path)
+		self.ftp.rmdir(name)
 
-	def _get_file_size(self, fullname):
-		lstatout = self.ftp.lstat(fullname)
-		return lstatout.st_size
+	@handle_permission_denied
+	def remove_by_file_name(self, name):
+		self.ftp.remove(name)
 
-	def delete(self, model, treeiter):
+	def walk(self, _dir, src, commands, remote_name, old_files):
```

Comment:
`_dir` is not a good name right here, prefixing an argument could too easily be confused with indicating that it's not used.